smart/tests/test_functions.py

```diff
@@ -0,0 +1,102 @@
+import numpy as np
+import pytest
+
+import astropy.units as u
+
+from sunpy.map import Map, all_coordinates_from_map, coordinate_is_on_solar_disk
+from sunpy.map.mapbase import GenericMap
+
+from smart.segmentation.map_processing import (
+    cosine_correction,
+    get_cosine_correction,
+    map_threshold,
+    smooth_los_threshold,
+)
+
+
+@pytest.fixture
+def hmi_nrt():
+    return "https://solmon.dias.ie/data/2024/06/06/HMI/fits/hmi.m_720s_nrt.20240606_230000_TAI.3.magnetogram.fits"
```

Comment:
Prob makes sense to get the data directly from JSOC e.g. http://jsoc.stanford.edu/data/hmi/fits/